### PR TITLE
[13.0][IMP] helpdesk_mgmt: Only show tickets of the user's helpdesk team with "User: Team tickets" permission.

### PR DESCRIPTION
**Pull request by victoralmau**
Only show tickets of the user's helpdesk team with "`User: Team tickets`" permission (related to https://github.com/OCA/helpdesk/pull/395#pullrequestreview-1229370921)

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT41172

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Why not doing this on the record rules, simply expanding them?

**Comment by victoralmau** [comment]
> Why not doing this on the record rules, simply expanding them?

What rule do you suggest?
Changing https://github.com/OCA/helpdesk/blob/13.0/helpdesk_mgmt/security/helpdesk_security.xml#L35 to limit according to helpdesk teams from user would be "incorrect" in some use cases.

**Comment by pedrobaeza** [comment]
Yes

**Comment by victoralmau** [comment]
Changes done, is this approach what you had in mind?

File: helpdesk_mgmt/security/helpdesk_security.xml [hunk 1]
```diff
@@ -32,15 +32,13 @@
             <field ref="model_helpdesk_ticket" name="model_id" />
             <field
                 name="domain_force"
-            >['|',('user_id','=',user.id),('user_id','=',False)]</field>
+            >[('team_id','in',user.helpdesk_team_ids.ids), '|',('user_id','=',user.id),('user_id','=',False)]</field>
```
Comment (pedrobaeza) [inline helpdesk_mgmt/security/helpdesk_security.xml]: This should cover also the case for team_id = False, but user_id = uid.

File: helpdesk_mgmt/security/helpdesk_security.xml [hunk 2]
```diff
@@ -32,15 +32,13 @@
...
             <field name="groups" eval="[(4, ref('group_helpdesk_user_own'))]" />
         </record>
         <record id="helpdesk_ticket_team_rule" model="ir.rule">
             <field name="name">Team Tickets</field>
             <field ref="model_helpdesk_ticket" name="model_id" />
-            <field
-                name="domain_force"
-            >['|',('team_id','in',user.helpdesk_team_ids.ids),('team_id','=',False)]</field>
+            <field name="domain_force">[('team_id','=',False)]</field>
```
Comment (pedrobaeza) [inline helpdesk_mgmt/security/helpdesk_security.xml]: This should match team_id = False and team_id = your team

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Include both changes in the migration script

**Comment by victoralmau** [comment]
> Include both changes in the migration script

Changes done.

File: helpdesk_mgmt/security/helpdesk_security.xml [hunk 3]
```diff
@@ -32,7 +32,7 @@
             <field ref="model_helpdesk_ticket" name="model_id" />
             <field
                 name="domain_force"
-            >['|',('user_id','=',user.id),('user_id','=',False)]</field>
+            >["&amp;","&amp;",('team_id','in',user.helpdesk_team_ids.ids),"|",('user_id','=',user.id),('user_id','=',False),"|",('user_id','=',user.id),('team_id','=',False)]</field>
```
Comment (pedrobaeza) [inline helpdesk_mgmt/security/helpdesk_security.xml]: Don't get complicated. The correct one should be:

```suggestion
            >["|", ('user_id', '=', user.id), '&amp;', ('user_id','=',False), ('team_id', 'in', user.helpdesk_team_ids.ids)]</field>
```

**Review by pedrobaeza — APPROVED** [review]
Let's fix the issue:

/ocabot merge nobump

**Pull request merged**